### PR TITLE
new templates for Automation

### PR DESCRIPTION
**Pull request by MGoedtel**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* New templates for creating Automation account and linking to new/existing Log Analytics workspace.
*
*

### CODE REVIEW

File: 101-automation/azuredeploy.parameters.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.parameters.json:1]: format the JSON, e.g. SHIFT+ALT+F in VS Code

File: 101-automation/azuredeploy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+            "value": "GEN-UNIQUE"
+        },
+        "location": {
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.parameters.json:8]: you will need a separate parameters file for AzureGov (see contribution guide)

File: 101-automation/azuredeploy.parameters.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+            "value": "eastus"
+        },
+        "resourcePermissions": {
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.parameters.json:11]: remove this and use a defaultValue in the template

File: 101-automation/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,207 @@
+{
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:1]: format the JSON

File: 101-automation/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    "dataRetention": {
+        "type": "int",
+        "defaultValue": 30,
+        "minValue": 7,
+        "maxValue": 730,
+        "metadata": {
+            "description": "Number of days to retain data."
+        }
+    },
+    "location": {
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:35]: add defaultValue of resourceGroup().location

File: 101-automation/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+            "type": "String",
+			"defaultValue": "AzureAutomationTutorial"
+        },
+        "sampleGraphicalRunbookDescription": {
+            "type": "String",
+			"defaultValue": " An example runbook that gets all the Resource Manager resources by using the Run As account (service principal)."
+        },
+        "sampleGraphicalRunbookContentUri": {
+            "type": "String",
+			"defaultValue": "https://eus2oaasibizamarketprod1.blob.core.windows.net/marketplace-runbooks/AzureAutomationTutorial.graphrunbook"
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:69]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts

File: 101-automation/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+                "name": "[parameters('sku')]",
+            },
+            "retentionInDays": "[parameters('dataRetention')]",
+            "features": {
+                "searchVersion": 1,
+                "legacy": 0,
+                "enableLogAccessUsingOnlyResourcePermissions": true
+            }
+        },
+	"resources": [
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:113]: remove this property

File: 101-automation/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+    {
+        "type": "Microsoft.Automation/automationAccounts",
+        "apiVersion": "2018-06-30",
+        "name": "[parameters('automationAccountName')]",
+        "location": "[parameters('automationAccountLocation')]",
+        "dependsOn": [
+		     "[concat('Microsoft.OperationalInsights/workspaces/', parameters('workspaceName'))]",
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:120]: remove concat and use resourceId

File: 101-automation/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+		],
+        "tags": {},
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:122]: remove all empty/null properties throughout the template (see CI results)

File: 101-automation/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+            }
+        },
+		"resources": [
+                {
+                    "type": "runbooks",
+                    "apiVersion": "2018-06-30",
+                    "name": "[parameters('sampleGraphicalRunbookName')]",
+                    "location": "[parameters('automationAccountLocation')]",
+                    "dependsOn": [
+                        "[concat('Microsoft.Automation/automationAccounts/', parameters('automationAccountName'))]"
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:135]: remove concat and use resourceId

File: 101-automation/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+                        }
+                    }
+                },
+                {
+                    "type": "runbooks",
+                    "apiVersion": "2018-06-30",
+                    "name": "[parameters('samplePowerShellRunbookName')]",
+                    "location": "[parameters('automationAccountLocation')]",
+                    "dependsOn": [
+                        "[concat('Microsoft.Automation/automationAccounts/', parameters('automationAccountName'))]"
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:155]: remove concat and use resourceId() won't comment on the rest

File: 101-automation/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+            "description": "Number of days to retain data."
+        }
+    },
+    "location": {
+        "type": "string",
+        "metadata": {
+            "description": "Specifies the location in which to create the workspace."
+        }
+    },
+	"resourcePermissions": {
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:41]: this is never used

File: 101-automation/azuredeploy.parameters.json (hunk 3 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.parameters.json:11]: Done

File: 101-automation/azuredeploy.json (hunk 4 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.json:1]: Done

File: 101-automation/azuredeploy.json (hunk 5 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.json:35]: Done

File: 101-automation/azuredeploy.parameters.json (hunk 2 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.parameters.json:8]: Done

File: 101-automation/azuredeploy.json (hunk 7 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.json:113]: Done

File: 101-automation/azuredeploy.json (hunk 9 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.json:122]: Done

File: 101-automation/azuredeploy.json (hunk 10 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.json:135]: Done

File: 101-automation/azuredeploy.json (hunk 12 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.json:41]: Done

File: 101-automation/azuredeploy.parameters.json (hunk 1 above)
Comment (MGoedtel) [reply 101-automation/azuredeploy.parameters.json:1]: Done

File: 101-automation/azuredeploy.json (hunk 6 above)
Comment (bmoore-msft) [reply 101-automation/azuredeploy.json]: This still needs to be done...

rest looks good, CI is working now...

Comment (MGoedtel) [reply 101-automation/azuredeploy.json]: Completed revising the link to scripts and using the _artifactsLocation method with assistance from Tom FitzMacken.

**Comment by MGoedtel** [comment]
All changes completed #sign-off

File: 101-automation/azuredeploy.json [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+            "type": "String",
+            "defaultValue": "AzureAutomationTutorialScript"
+        },
+        "samplePowerShellRunbookDescription": {
+            "type": "String",
+            "defaultValue": " An example runbook that gets all the Resource Manager resources by using the Run As account (service principal)."
+        },
+        "samplePowerShellRunbookContentUri": {
+            "type": "String",
+            "defaultValue": "https://eus2oaasibizamarketprod1.blob.core.windows.net/marketplace-runbooks/AzureAutomationTutorial.ps1"
```
Comment (bmoore-msft) [inline 101-automation/azuredeploy.json:76]: I'm still seeing all the uri's pointing to the marketplace storage - not sure if the commit didn't get pushed or something else, but be sure to get all of them.

Comment (MGoedtel) [reply 101-automation/azuredeploy.json]: Noticed that my commits weren't completing successfully due to tip of my current branch behind.  After resolving that it appears the PR is up-to-date with the changes.

**Comment by MGoedtel** [comment]
@bmoore-msft - It looks like the validation workflow doesn't like when there are two instances of resourceGroup.Location used in a template.  Is this true or am I misinterpreting the results?  The template is using that for the automationAccountLocation parameter and LA Workspace location parameter.  Please advise if there is action on my part to resolve this in order to maintain this cfg in the template.  Thanks.

**Comment by MGoedtel** [comment]
Assuming the changes you made per my earlier comment indicate the templates are meeting the quality bar and can be merged.  #sign-off

**Pull request merged**